### PR TITLE
Implement heatmap features, initial cleanup and redesign

### PR DESCRIPTION
**Pull request by jperson**

### CODE REVIEW

File: pages/heatmap/basic.vue [hunk 1]
```diff
@@ -1,105 +1,282 @@
...
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
```
Comment (jperson) [inline pages/heatmap/basic.vue:47]: need to change the formatter settings to use longer line widths for js code

File: store/modules/heatmap/index.js [hunk 2]
```diff
@@ -4,16 +4,29 @@ import getters from "./getters";
...
+    "Contraction Velocity",
+    "Relaxation Velocity",
+  ],
+  heatmap_display_idx: 0,
   heatmap_display_min_max: [],
   heatmap_display_user_min_max: {},
   heatmap_autoscale: false,
-  heatmap_options_array: [],
-  heatmap_options_idx: null,
+  heatmap_options_array: ["Warm", "Cool", "Blue/Red", "Purple/Green"],
```
Comment (jperson) [inline store/modules/heatmap/index.js:28]: delete this, the options get set from the gradient store values now

File: store/modules/heatmap/mutations.js [hunk 3]
```diff
@@ -33,4 +33,9 @@ export default {
   set_heatmap_options_gradient(state, new_value) {
     state.heatmap_options_gradient = new_value;
   },
+  set_metric_data(state, new_value) {
```
Comment (jperson) [inline store/modules/heatmap/mutations.js:36]: this is for testing right now, we need to design a better way to handle the streaming data, probably in it's own store other components read from with a compute method

File: store/modules/heatmap/index.js (hunk 2 above)
Comment (luciipak) [reply store/modules/heatmap/index.js:28]: Got it!

File: pages/heatmap/basic.vue [hunk 4]
```diff
@@ -1,105 +1,122 @@
...
-        { color: "#ffff8c", offset: "100%" },
-      ],
+    const data = [
+      [[], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], []],
+      [[], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], []],
+      [[], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], []],
+      [[], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], []],
+      [[], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], []],
+      [[], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], []],
     ];
```
Comment (luciipak) [inline pages/heatmap/basic.vue:29]: I kind of miss the 100 lines of '[]'

**Review by luciipak — APPROVED** [review]
Looks really clean, I'm looking forward to seeing how it. Will we still need the input-dropdown component?

**Comment by jperson** [comment]
> I'm looking forward to seeing how it looks! Will we still need the input-dropdown component?

I left it for now, I think it's kind of confusing how it works and prefer a normal dropdown box but there might be other uses for it. The SelectDropDown will need some tweaks to make the styling match the rest of the app, but we can get to that later

**Comment by luciipak** [comment]
> > I'm looking forward to seeing how it looks! Will we still need the input-dropdown component?
> 
> I left it for now, I think it's kind of confusing how it works and prefer a normal dropdown box but there might be other uses for it. The SelectDropDown will need some tweaks to make the styling match the rest of the app, but we can get to that later

Yeah, that's no problem

**Pull request merged**